src/finn/custom_op/fpgadataflow/rtl/thresholding_rtl.py

```diff
@@ -0,0 +1,776 @@
+        code_gen_dict["$DEEP_PIPELINE$"] = [str(deep_pipeline)]
+        return code_gen_dict
+
+    def get_rtl_file_list(self):
+        """Thresholding binary search RTL file list"""
+        return [
+            "axilite_if.v",
+            "thresholding.sv",
+            "thresholding_axi.sv",
+            "thresholding_template_wrapper.v",
+        ]
+
+    def get_rtl_file_paths(self):
+        """Get full path of all RTL files"""
+        rtl_root_dir = os.environ["FINN_ROOT"] + "/finn-rtllib/thresholding/hdl/"
+        rtl_file_list = self.get_rtl_file_list()
+        rtl_file_paths = [rtl_root_dir + file for file in rtl_file_list]
+        return rtl_file_paths
+
+    def get_rtl_template_data(self, path):
+        """Return RTL file contents as a template"""
+        with open(path, "r") as f:
+            template = f.read()
+        return template
+
+    def fill_in_rtl_template_data(self, replace_dict, template_data):
+        """Use attribute values to finn in RTL template placeholders"""
+        template_data_cp = template_data
+        for key in replace_dict:
+            replacement_line = "\n".join(replace_dict[key])
+            template_data_cp = template_data_cp.replace(key, replacement_line)
+        return template_data_cp
+
+    def dump_rtl_data(self, dest_dir, filename, data):
+        """Dump filled-in-template RTL files for future synthesis step"""
+        # when generating template files, handle a special case:
+        # if the filename contains the word "template", replace that
+        # with the node name to distinguish between instances
+        filename = filename.replace("template", self.onnx_node.name)
+        with open(os.path.join(dest_dir, filename), "w") as f:
+            f.write(data)
+        return
+
+    def generate_hdl(self, model):
+        """Prepare HDL files from templates for synthesis"""
+        # Generate a dictionary of values to put in RTL template
+        code_gen_dict = self.prepare_codegen_rtl_values(model)
+
+        # Retrieve the destination directory for the final RTL files
+        code_gen_dir = self.get_nodeattr("code_gen_dir_ipgen")
+
+        for rtl_file_path in self.get_rtl_file_paths():
+            # read in original RTL template file
+            template_data = self.get_rtl_template_data(rtl_file_path)
+            # apply code generation to templates
+            data = self.fill_in_rtl_template_data(code_gen_dict, template_data)
+            # dump filled-in template to destination directory for compilation
+            file_only_path = rtl_file_path.split("/")[-1]
+            self.dump_rtl_data(code_gen_dir, file_only_path, data)
+
+        # Before we return - set the 'gen_top_module' attribute for use later
+        # by PyVerilator and IPI generation
+        self.set_nodeattr("gen_top_module", code_gen_dict["$TOP_MODULE$"][0])
+        return
+
+    def code_generation_ipgen(self, model, fpgapart, clk):
```

Comment:
If you move this code block
`# set ipgen_path and ip_path so that HLS-Synth transformation`
`# and stich_ip transformation do not complain`
`# i.e. during the HLSSynthIP() transformation`
`code_gen_dir = self.get_nodeattr("code_gen_dir_ipgen")`
`self.set_nodeattr("ipgen_path", code_gen_dir)`
`self.set_nodeattr("ip_path", code_gen_dir)`
to `generate_hdl()`, then you can also remove this function. By default `code_generation_ipgen` executes `generate_hdl`, as defined in rtlbackend: https://github.com/Xilinx/finn/blob/refactor/rtl_integration/src/finn/custom_op/fpgadataflow/rtlbackend.py#L56